src/python/pants/backend/python/util_rules/pex_from_targets.py

```diff
@@ -657,5 +663,69 @@ async def get_requirements_pex(request: RequirementsPexRequest, setup: PythonSet
     return pex_request
 
 
+@dataclass(frozen=True)
+class PexReqsRequest:
+    addresses: Addresses
+    interpreter_constraints: InterpreterConstraints | None = None
+
+
+
+@dataclass(frozen=True)
+class PexReqs:
+    requirements: Lockfile | PexRequirements
+    pexes: Iterable[Pex]
```

Comment:
Most iterables would not be frozen and hashable, so consider using `tuple` here instead.